Code/src/main/java/nl/utwente/group10/ui/components/blocks/Block.java

```diff
@@ -32,41 +38,55 @@ public Block(CustomUIPane pane) {
         try {
             output = new OutputAnchor(this, pane);
         } catch (IOException e) {
-            // TODO Find a good way to handle this
             e.printStackTrace();
         }
 
-        parentPane.selectedBlockProperty().addListener(event -> {
-            if (parentPane.getSelectedBlock().isPresent() && this.equals(parentPane.getSelectedBlock().get())) {
-                this.getStyleClass().add("selected");
-            } else {
-                this.getStyleClass().removeAll("selected");
-            }
-        });
+        parentPane.selectedBlockProperty()
+                .addListener(
+                        event -> {
+                            if (parentPane.getSelectedBlock().isPresent()
+                                    && this.equals(parentPane
+                                            .getSelectedBlock().get())) {
+                                this.getStyleClass().add("selected");
+                            } else {
+                                this.getStyleClass().removeAll("selected");
+                            }
+                        });
+
+        this.addEventHandler(MouseEvent.MOUSE_CLICKED, this::handleMouseEvent);
 
-        this.addEventHandler(MouseEvent.MOUSE_CLICKED, this::select);
+        this.createContextMenu();
     }
 
-    /** Sets this block as the selected block. */
-    private void select(MouseEvent mouseEvent) {
-        parentPane.setSelectedBlock(this);
+    private void createContextMenu() {
```

Comment:
If you mean the createContextMenu then I left it out on purpose, the idea is that block implementations with unique features can override the method to call their own specific CircleMenu implementation. (I imagine stuff like ValueBlock being able to adjust it's values etc.)